static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,279 @@
+    }
+  );
+}
+
+type ResetAction = {
+  type: 'reset';
+};
+
+type SetCommittersError = {
+  payload: {
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'set error';
+};
+
+type StartLoadingCommitters = {
+  payload: {
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'start loading';
+};
+
+type AddCommitters = {
+  payload: {
+    committers: Committer[];
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'add committers';
+};
+
+export type CommittersState = Record<
+  string,
+  {
+    committers: Committer[];
+    committersError: boolean;
+    committersLoading: boolean;
+  }
+>;
+
+export type CommittersAction =
+  | ResetAction
+  | AddCommitters
+  | StartLoadingCommitters
+  | SetCommittersError;
+
+export function committersReducer(state, action: CommittersAction): CommittersState {
+  switch (action.type) {
+    case 'add committers': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: action.payload.committers,
+          committersLoading: false,
+          committersError: undefined,
+        },
+      };
+    }
+
+    case 'start loading': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: true,
+          committersError: undefined,
+        },
+      };
+    }
+    case 'set error': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: false,
+          committersError: true,
+        },
+      };
+    }
+    case 'reset': {
+      return {};
+    }
+    default: {
+      exhaustive(action);
+      return state;
+    }
+  }
+}
+
+export const CommittersContext = createContext<
+  [CommittersState, React.Dispatch<CommittersAction>] | null
+>(null);
+
+interface CommittersContextProviderProps {
+  children: React.ReactNode;
+  initialState?: CommittersState;
+}
+
+export function CommittersProvider(props: CommittersContextProviderProps) {
+  const contextValue = useReducer(committersReducer, props.initialState ?? {});
+
+  return (
+    <CommittersContext.Provider value={contextValue}>
+      {props.children}
+    </CommittersContext.Provider>
+  );
+}
+
+export function useCommitters(): [CommittersState, React.Dispatch<CommittersAction>] {
```

Comment:
we are writing hooks in the folder `sentry/utils/*`. how we can make them more discoverable?

Reply:
Hmm, idk. This hook is really just like 10 lines of code. Do you think it's worth having it in a separate file?
```
import {ConfigContext}...

export function useConfig(): [Config, React.Dispatch<ConfigAction>] {
  const context = useContext(ConfigContext);

  if (!context) {
    throw new Error('useConfig called outside of ConfigProvider');
  }

  return context;
}
```

Reply:
no strong feelings here. I just think it would be more discoverable if it was in the `sentry/utils` folder